rust/agents/relayer/src/msg/processor.rs

```diff
@@ -117,17 +97,10 @@ impl MessageProcessor {
             tokio::time::sleep(Duration::from_secs(1)).await;
             return Ok(());
         };
-
-        // Skip if for different domain.
-        if message.destination != self.destination_mailbox.domain().id() {
-            debug!(
-                id=?message.id(),
-                destination=message.destination,
-                nonce=message.nonce,
-                "Message destined for other domain, skipping");
-            self.message_nonce += 1;
-            return Ok(());
-        }
+        self.metrics
+            .get(message.destination)
+            .unwrap()
```

Comment:
I believe this will panic if the message's destination is one that isn't supported by this relayer. We should either gracefully handle the None case, or put this above `send_channel.send(submit_args)?;` on line 144 (which would mean that we'd filter using the whitelist/blacklist before updating the metrics, which is maybe desirable or maybe not?), or change the `get` function in `MessageProcessorMetrics` to just make the `with_label_values` call each time it's called. I don't feel strongly which way